rclcpp/include/rclcpp/copy_all_parameter_values.hpp

```diff
@@ -77,6 +81,28 @@ copy_all_parameter_values(
   }
 }
 
+/// Load a list of parameters from a yaml parameter file.
+/**
+ * \param[in] yaml_name The name of the yaml file that needs to be loaded.
+ * \param[in] node_interfaces The list of variadic NodeInterfaces taken as a template to set parameters for
+ */
+template<typename NodeT>
+std::vector<rcl_interfaces::msg::SetParametersResult>
+load_parameters(
+  const std::string & yaml_filepath, NodeT node_interface)
```

Comment:
I would prefer if we didn't use a template here, but rather the `NodeInterfaces` class or directly passing the two required interface (`node_base_interface` and `node_parameters_interface`)

Reply:
that would be better for compile time.